### PR TITLE
LANG-1127: Rule alternative implementation

### PR DESCRIPTION
**Pull request by chonton**
Use annotation to set new TimeZone/Locale values per test method.  No synchronization yet.

### CODE REVIEW

**Comment by britter** [comment]
Look very good to me, expect two changes. I'll comment inline.

File: src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java [hunk 1]
```diff
@@ -111,43 +108,39 @@ public void test_getInstance_String_Locale() {
...
-        locale.setLocale(Locale.US);
-
         final FastDateFormat format1 = FastDateFormat.getDateInstance(FastDateFormat.FULL, Locale.GERMANY);
         final FastDateFormat format2 = FastDateFormat.getDateInstance(FastDateFormat.FULL);
         Locale.setDefault(Locale.GERMANY);
         final FastDateFormat format3 = FastDateFormat.getDateInstance(FastDateFormat.FULL);
 
         assertSame(Locale.GERMANY, format1.getLocale());
-        assertSame(Locale.US, format2.getLocale());
+        assertEquals(Locale.US, format2.getLocale());
```
Comment (britter) [inline src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java]: Has this been changed intentionally?

File: src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java [hunk 2]
```diff
@@ -111,43 +108,39 @@ public void test_getInstance_String_Locale() {
...
-        locale.setLocale(Locale.US);
-
         final FastDateFormat format1 = FastDateFormat.getDateTimeInstance(FastDateFormat.FULL, FastDateFormat.FULL, Locale.GERMANY);
         final FastDateFormat format2 = FastDateFormat.getDateTimeInstance(FastDateFormat.FULL, FastDateFormat.FULL);
         Locale.setDefault(Locale.GERMANY);
         final FastDateFormat format3 = FastDateFormat.getDateTimeInstance(FastDateFormat.FULL, FastDateFormat.FULL);
 
         assertSame(Locale.GERMANY, format1.getLocale());
-        assertSame(Locale.US, format2.getLocale());
+        assertEquals(Locale.US, format2.getLocale());
```
Comment (britter) [inline src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java]: Has this been changed intentionally?

Comment (chonton) [inline src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java]: Yes,  Locales are unlike enumerations which have a single instance per value.  Locale.US != new Locale("en", "US").

File: src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java (hunk 1 above)
Comment (chonton) [inline src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java]: Yes

**Pull request merged**